libraries/ride-fuzzer-lib/src/main/java/com/adobe/ride/libraries/fuzzer/MetadataFuzzer.java

```diff
@@ -140,45 +148,142 @@ private void initializeFuzzer(String serviceName, ModelObject objectToBeFuzzed,
 
     // Populate global properties
     this.entity = objectToBeFuzzed;
+    this.entity.buildValidModelInstance();
+
     this.requestMethod = method;
     this.serviceName = serviceName;
-    this.properties = this.entity.getModelProperties();
-    this.instance = this.entity.getObjectMetadata();
+    this.properties = ModelObject.getObjectNodeProperties(entity, this.entity.getModel());
     this.requestMethod = method;
     this.contentType = contentType;
     this.requestBuilder = requestBuilder;
+    
+    if (requestBuilder != null) {
+      this.requestBuilder = requestBuilder;
+    } else {
+      this.requestBuilder = RestApiController.getRequestBuilder(false);
+      this.requestBuilder.addHeader("Content-Type", "application/json");
```

Comment:
Optimization improvement.  Creating the Rest Assured RequestSpecBuilder in the MetaDataFuzzer for one-time creation in run, instead of in the MetadataEngine.